aide_design/cdc_functions.py

```diff
@@ -36,31 +34,48 @@ def _DiamTubeAvail(en_tube_series = True):
 #         return (1/16)*u.inch
 #==============================================================================
 
-NU_WATER = 1*(u.mm**2/u.s)
-
-@u.wraps(u.m**2/u.s, [u.kg/u.m**3], False)
-def _nu_alum(conc_alum):
-    nu = (1 + (4.255 * 10**-6) * conc_alum**2.289) * NU_WATER
-    return nu.to(u.mm**2/u.s)
-    
+@u.wraps(u.m**2/u.s, [u.kg/u.m**3, u.degK], False)
+def viscosity_kinematic_alum(conc_alum, temp):
+    """Return the dynamic viscosity of water at a given temperature.
+    
+    If given units, the function will automatically convert to Kelvin.
+    If not given units, the function will assume Kelvin.
+    This function assumes that the temperature dependence can be explained 
+    based on the effect on water and that there is no confounding effect from 
+    the coagulant.
+    """
+    nu = (1 + (4.255 * 10**-6) * conc_alum**2.289) * pc.viscosity_kinematic(temp).magnitude
     return nu
 
 
-@u.wraps(u.m**2/u.s, [u.kg/u.m**3], False)   
-def _nu_pacl(conc_pacl):
-    nu = (1 + (2.383 * 10**-5) * conc_pacl**1.893) * NU_WATER
-    return nu.to(u.mm**2/u.s)
+@u.wraps(u.m**2/u.s, [u.kg/u.m**3, u.degK], False)   
+def viscosity_kinematic_pacl(conc_pacl, temp):
+    """Return the dynamic viscosity of water at a given temperature.
     
+    If given units, the function will automatically convert to Kelvin.
+    If not given units, the function will assume Kelvin.
+    This function assumes that the temperature dependence can be explained 
+    based on the effect on water and that there is no confounding effect from 
+    the coagulant.
+    """
+    nu = (1 + (2.383 * 10**-5) * conc_pacl**1.893) * pc.viscosity_kinematic(temp).magnitude
     return nu
 
 
-@u.wraps(u.m**2/u.s, [u.kg/u.m**3, None], False)    
-def _nu_chem(conc_chem, en_chem):
-    if en_chem == 0:
-        return _nu_alum(conc_chem)
-        return _nu_pacl(conc_chem)
-    else:
-        return NU_WATER
+@u.wraps(u.m**2/u.s, [u.kg/u.m**3, u.degK, None], False)    
+def viscosity_kinematic_chem(conc_chem, temp, en_chem):
+     """Return the dynamic viscosity of water at a given temperature.
+    
+    If given units, the function will automatically convert to Kelvin.
+    If not given units, the function will assume Kelvin.
+    """
+     if en_chem == 0:
```

Comment:
en_chem (if you choose to do the enum route) should be defined in expertinputs or some such place. That way, the code would read `if en_chem == expert_inputs.en_chem.PACL`... or you could import en_chem so it reads just: `if en_chem == en_chem.PACL`